source/config.go

```diff
@@ -24,48 +23,11 @@ import (
 const (
 	// ConfigKeyPollingPeriod is the config name for the S3 CDC polling period
 	ConfigKeyPollingPeriod = "pollingPeriod"
-
-	// DefaultPollingPeriod is the value assumed for the pooling period when the
-	// config omits the polling period parameter
-	DefaultPollingPeriod = "1s"
 )
 
 // Config represents source configuration with S3 configurations
 type Config struct {
 	config.Config
-	PollingPeriod time.Duration
-}
-
-// Parse attempts to parse the configurations into a Config struct that Source could utilize
-func Parse(cfg map[string]string) (Config, error) {
-	common, err := config.Parse(cfg)
-	if err != nil {
-		return Config{}, err
-	}
-
-	pollingPeriodString, exists := cfg[ConfigKeyPollingPeriod]
-	if !exists || pollingPeriodString == "" {
-		pollingPeriodString = DefaultPollingPeriod
-	}
-	pollingPeriod, err := time.ParseDuration(pollingPeriodString)
-	if err != nil {
-		return Config{}, fmt.Errorf(
-			"%q config value should be a valid duration",
-			ConfigKeyPollingPeriod,
-		)
-	}
-	if pollingPeriod <= 0 {
-		return Config{}, fmt.Errorf(
-			"%q config value should be positive, got %s",
-			ConfigKeyPollingPeriod,
-			pollingPeriod,
-		)
-	}
-
-	sourceConfig := Config{
-		Config:        common,
-		PollingPeriod: pollingPeriod,
-	}
-
-	return sourceConfig, nil
+	// polling period for the CDC mode, formatted as a time.Duration string.
+	PollingPeriod time.Duration `json:"pollingPeriod" default:"1s"`
```

Comment:
I see we don't enforce positive durations like we did previously, I expect the validations don't work on a duration. Would be nice if we add support for durations, so we could simply add `validate:"gt=0"` here 😉 Not a blocker!

Reply:
> we could make this possible from the SDK and update Greater-Than to compare durations too

That's what I meant. But as I said, no need to block this PR because of this, it's a separate issue that I just noticed here.

Reply:
Opened an issue: https://github.com/ConduitIO/conduit-connector-sdk/issues/54.